tests/python/sdk/test_projects.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (C) 2022 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import io
+import os
+from logging import Logger
+from pathlib import Path
+from typing import Tuple
+
+import pytest
+from cvat_sdk import Client, models
+from cvat_sdk.api_client import exceptions
+from cvat_sdk.core.proxies.projects import Project
+from cvat_sdk.core.proxies.tasks import ResourceType, Task
+from cvat_sdk.core.utils import filter_dict
+
+from .util import make_pbar
+
+
+class TestProjectUsecases:
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        tmp_path: Path,
+        fxt_login: Tuple[Client, str],
+        fxt_logger: Tuple[Logger, io.StringIO],
+        fxt_stdout: io.StringIO,
+    ):
+        self.tmp_path = tmp_path
+        logger, self.logger_stream = fxt_logger
+        self.stdout = fxt_stdout
+        self.client, self.user = fxt_login
+        self.client.logger = logger
+
+        api_client = self.client.api_client
+        for k in api_client.configuration.logger:
+            api_client.configuration.logger[k] = logger
+
+    @pytest.fixture
+    def fxt_new_task(self, fxt_image_file: Path):
+        task = self.client.tasks.create_from_data(
+            spec={
+                "name": "test_task",
+                "labels": [{"name": "car"}, {"name": "person"}],
+            },
+            resource_type=ResourceType.LOCAL,
+            resources=[str(fxt_image_file)],
+            data_params={"image_quality": 80},
+        )
+
+        return task
+
+    @pytest.fixture
+    def fxt_task_with_shapes(self, fxt_new_task: Task):
+        fxt_new_task.set_annotations(
+            models.LabeledDataRequest(
+                shapes=[
+                    models.LabeledShapeRequest(
+                        frame=0,
+                        label_id=fxt_new_task.labels[0].id,
+                        type="rectangle",
+                        points=[1, 1, 2, 2],
+                    ),
+                ],
+            )
+        )
+
+        return fxt_new_task
+
+    @pytest.fixture
+    def fxt_new_project(self):
+        project = self.client.projects.create(
+            spec={
+                "name": "test_project",
+                "labels": [{"name": "car"}, {"name": "person"}],
+            },
+        )
+
+        return project
+
+    @pytest.fixture
+    def fxt_empty_project(self):
+        project = self.client.projects.create(spec={"name": "test_project"})
+
+        return project
+
+    @pytest.fixture
+    def fxt_project_with_shapes(self, fxt_empty_project: Project, fxt_task_with_shapes: Task):
+        fxt_empty_project.update(
```

Comment:
IMO, fixtures shouldn't modify other fixtures, because otherwise you can get confusing behavior:

```python
def test_whatever(fxt_empty_project, fxt_project_with_shapes):
    # a project with shapes can't be empty, but fxt_empty_project is the same object as fxt_project_with_shapes
```

Also, in this particular case `fxt_empty_project` is just unnecessary. It's just one line of code, it doesn't need to be a fixture.